Mathlib/CategoryTheory/Functor/FullyFaithful.lean

```diff
@@ -79,35 +69,23 @@ theorem mapIso_injective (F : C ⥤ D) [Faithful F] :
   Iso.ext (map_injective F (congr_arg Iso.hom h : _))
 #align category_theory.functor.map_iso_injective CategoryTheory.Functor.mapIso_injective
 
+theorem map_surjective (F : C ⥤ D) [Full F] :
+    Function.Surjective (F.map : (X ⟶ Y) → (F.obj X ⟶ F.obj Y)) :=
+  Full.map_surjective
+#align category_theory.functor.map_surjective CategoryTheory.Functor.map_surjective
+
 /-- The specified preimage of a morphism under a full functor. -/
 @[pp_dot]
-def preimage (F : C ⥤ D) [Full F] (f : F.obj X ⟶ F.obj Y) : X ⟶ Y :=
-  Full.preimage.{v₁, v₂} f
+noncomputable def preimage (F : C ⥤ D) [Full F] (f : F.obj X ⟶ F.obj Y) : X ⟶ Y :=
+  (F.map_surjective f).choose
 #align category_theory.functor.preimage CategoryTheory.Functor.preimage
 
 @[simp]
 theorem image_preimage (F : C ⥤ D) [Full F] {X Y : C} (f : F.obj X ⟶ F.obj Y) :
```

Comment:
This should be called `map_preimage` instead...
Not for this PR though.

Reply:
I had also spotted this bad name for a certain time... This is a good occasion to change it!